flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskExecutorPartitionLifecycleTest.java

```diff
@@ -276,4 +409,69 @@ private TestingTaskExecutor createTestingTaskExecutor(TaskManagerServices taskMa
 				null),
 			new TestingFatalErrorHandler());
 	}
+
+	private static class TestJobAwareShuffleEnvironment implements JobAwareShuffleEnvironment<ResultPartitionWriter, InputGate> {
+
+		private final Function<JobID, Boolean> hasPartitionsOccupyingLocalResourcesFunction;
+		private Consumer<JobID> listener = null;
+
+		private TestJobAwareShuffleEnvironment(Function<JobID, Boolean> hasPartitionsOccupyingLocalResourcesFunction) {
+			this.hasPartitionsOccupyingLocalResourcesFunction = hasPartitionsOccupyingLocalResourcesFunction;
+		}
+
+		@Override
+		public int start() throws IOException {
+			return 0;
+		}
+
+		@Override
+		public Collection<ResultPartitionWriter> createResultPartitionWriters(JobID jobId, String taskName, ExecutionAttemptID executionAttemptID, Collection<ResultPartitionDeploymentDescriptor> resultPartitionDeploymentDescriptors, MetricGroup outputGroup, MetricGroup buffersGroup) {
```

Comment:
too long line, break into multiple lines?

Reply:
wouldn't worry about it since the methods aren't doing anything.

Reply:
:)